pkg/k8s/k8siface.go

```diff
@@ -100,6 +101,7 @@ func (k *k8sIface) AddGWLabelOnNode(nodeName string) error {
 		}
 
 		labels[SubmarinerGatewayLabel] = "true"
+		labels[InfraNodeLabel] = ""
```

Comment:
Since this node isnt managed by us, should we really add this label?

Reply:
Even I wasn't 100% sure of what would be right way to do this.  Alternately, should we install gw only on nodes that have infra label, at least OCP clusters? Any node running gateway should be treated as infra node but since we're selecting an existing node, as you said, we're unlikely to be owner of it.